diff --git a/boost/context/detail/config.hpp b/boost/context/detail/config.hpp
index 06737fdf9..9bb9ce8cc 100644
--- a/boost/context/detail/config.hpp
+++ b/boost/context/detail/config.hpp
@@ -119,11 +119,6 @@ static constexpr std::size_t cacheline_length{ 64 };
 static constexpr std::size_t prefetch_stride{ 4 * cacheline_length };
 #endif
 
-#if defined(__GLIBCPP__) || defined(__GLIBCXX__)
-// GNU libstdc++ 3
-#  define BOOST_CONTEXT_HAS_CXXABI_H
-#endif
-
 #if defined( BOOST_CONTEXT_HAS_CXXABI_H )
 # include <cxxabi.h>
 #endif
