diff --git a/gdb/event-top.c b/gdb/event-top.c
index 3138e8c..f86feb4 100644
--- a/gdb/event-top.c
+++ b/gdb/event-top.c
@@ -80,6 +80,11 @@ static void async_sigtstp_handler (gdb_client_data);
 #endif
 static void async_sigterm_handler (gdb_client_data arg);
 
+#ifndef HAVE_STRSIGNAL
+/* While libiberty provides a fallback, it doesn't provide a prototype.  */
+extern "C" char *strsignal (int __signo);
+#endif
+
 /* Instead of invoking (and waiting for) readline to read the command
    line and pass it back for processing, we use readline's alternate
    interface, via callback functions, so that the event loop can react
