diff --git a/meson.build b/meson.build
index a39eac5..97c7c7e 100644
--- a/meson.build
+++ b/meson.build
@@ -147,9 +147,6 @@ subdir('po')
 subdir('proxy')
 subdir('tls')
 
-# Will automatically pick it up from the cross file if defined
-gnome.post_install(gio_querymodules: gio_module_dir)
-
 summary({
   'GnuTLS support': giomodules.contains('giognutls'),
   'OpenSSL support': giomodules.contains('gioopenssl'),
diff --git a/proxy/meson.build b/proxy/meson.build
index 266cea4..87dd5d7 100644
--- a/proxy/meson.build
+++ b/proxy/meson.build
@@ -9,5 +9,3 @@ endif
 if environment_proxy_enabled
   subdir('environment')
 endif
-
-subdir('tests')
diff --git a/tls/meson.build b/tls/meson.build
index e76e766..b8103cd 100644
--- a/tls/meson.build
+++ b/tls/meson.build
@@ -13,5 +13,3 @@ if backends.contains('openssl')
 
   subdir('openssl')
 endif
-
-subdir('tests')
