diff -ruN jonaski-gstreamer-69bd121.orig/plugins/elements/gstfilesink.c jonaski-gstreamer-69bd121/plugins/elements/gstfilesink.c
--- jonaski-gstreamer-69bd121.orig/plugins/elements/gstfilesink.c	2018-09-03 20:50:33.000000000 +0200
+++ jonaski-gstreamer-69bd121/plugins/elements/gstfilesink.c	2018-09-03 21:21:43.717335566 +0200
@@ -771,10 +771,10 @@
     guint size = 0;
     gst_buffer_list_foreach (buffer_list, accumulate_size, &size);
 
-    GST_DEBUG_OBJECT (sink,
-        "Queueing buffer list of %u bytes (%u buffers) at offset %"
-        G_GSIZE_FORMAT, size, num_buffers,
-        sink->current_pos + sink->current_buffer_size);
+    //GST_DEBUG_OBJECT (sink,
+      //  "Queueing buffer list of %u bytes (%u buffers) at offset %"
+        //G_GSIZE_FORMAT, size, num_buffers,
+        //sink->current_pos + sink->current_buffer_size);
 
     for (i = 0; i < num_buffers; ++i)
       gst_buffer_list_add (sink->buffer,
@@ -826,10 +826,10 @@
           gst_file_sink_render_buffers (filesink, &buffer, 1, &n_mem, n_mem,
           gst_buffer_get_size (buffer));
   } else if (n_mem > 0) {
-    GST_DEBUG_OBJECT (filesink,
-        "Queueing buffer of %" G_GSIZE_FORMAT " bytes at offset %"
-        G_GSIZE_FORMAT, gst_buffer_get_size (buffer),
-        filesink->current_pos + filesink->current_buffer_size);
+    //GST_DEBUG_OBJECT (filesink,
+      //  "Queueing buffer of %" G_GSIZE_FORMAT " bytes at offset %"
+        //G_GSIZE_FORMAT, gst_buffer_get_size (buffer),
+        //filesink->current_pos + filesink->current_buffer_size);
 
     filesink->current_buffer_size += gst_buffer_get_size (buffer);
     gst_buffer_list_add (filesink->buffer, gst_buffer_ref (buffer));
