diff --git a/configure.ac b/configure.ac
index 5a2f76c..44b3340 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5,19 +5,21 @@ AC_PREREQ([2.63])
 AC_INIT([libbs2b], [3.1.0], [boris_mikhaylov@users.sourceforge.net])
 AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE([1.10.1 -Wall foreign subdir-objects
-                  dist-zip dist-bzip2 dist-lzma])
+                  dist-zip dist-bzip2 dist-xz])
 AC_CONFIG_SRCDIR([src/bs2b.h])
 
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT
 PKG_PROG_PKG_CONFIG
 
 # Checks for libraries.
 PKG_CHECK_EXISTS([sndfile], [], [
-    AC_MSG_ERROR(Please install libsndfile.)
-])
+    AC_MSG_WARN(libsndfile not found. Program building disabled.)
+], [enable_programs=yes])
+
+AM_CONDITIONAL(PROGRAMS, test "x$enable_programs" = "xyes")
 
 # Checks for header files.
 AC_CHECK_HEADERS([fcntl.h malloc.h string.h])
diff --git a/src/Makefile.am b/src/Makefile.am
index 414aeec..cb46f1a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,11 +1,6 @@
 lib_LTLIBRARIES = \
 	libbs2b.la
 
-bin_PROGRAMS = \
-	bs2bconvert \
-	bs2bstream
-
-
 bs2b_HEADERS = \
 	bs2b.h \
 	bs2bclass.h \
@@ -24,6 +19,11 @@ libbs2b_la_SOURCES = \
 	bs2b.c \
 	bs2bclass.cpp
 
+if PROGRAMS
+bin_PROGRAMS = \
+	bs2bconvert \
+	bs2bstream
+
 bs2bconvert_LDADD = \
 	libbs2b.la
 
@@ -38,3 +38,4 @@ bs2bstream_LDADD = \
 
 bs2bstream_SOURCES = \
 	bs2bstream.c
+endif
diff --git a/src/bs2b.c b/src/bs2b.c
index 623ca77..b87680c 100644
--- a/src/bs2b.c
+++ b/src/bs2b.c
@@ -350,12 +350,6 @@ void bs2b_cross_feed_d( t_bs2bdp bs2bdp, double *sample, int n )
 		{
 			cross_feed_d( bs2bdp, sample );
 			
-			/* Clipping of overloaded samples */
-			if( sample[ 0 ] >  1.0 ) sample[ 0 ] =  1.0;
-			if( sample[ 0 ] < -1.0 ) sample[ 0 ] = -1.0;
-			if( sample[ 1 ] >  1.0 ) sample[ 1 ] =  1.0;
-			if( sample[ 1 ] < -1.0 ) sample[ 1 ] = -1.0;
-
 			sample += 2;
 		} /* while */
 	} /* if */
@@ -374,12 +368,6 @@ void bs2b_cross_feed_dbe( t_bs2bdp bs2bdp, double *sample, int n )
 
 			cross_feed_d( bs2bdp, sample );
 
-			/* Clipping of overloaded samples */
-			if( sample[ 0 ] >  1.0 ) sample[ 0 ] =  1.0;
-			if( sample[ 0 ] < -1.0 ) sample[ 0 ] = -1.0;
-			if( sample[ 1 ] >  1.0 ) sample[ 1 ] =  1.0;
-			if( sample[ 1 ] < -1.0 ) sample[ 1 ] = -1.0;
-
 			#ifndef WORDS_BIGENDIAN
 			int64swap( ( uint32_t * )sample );
 			int64swap( ( uint32_t * )( sample + 1 ) );
@@ -403,12 +391,6 @@ void bs2b_cross_feed_dle( t_bs2bdp bs2bdp, double *sample, int n )
 
 			cross_feed_d( bs2bdp, sample );
 
-			/* Clipping of overloaded samples */
-			if( sample[ 0 ] >  1.0 ) sample[ 0 ] =  1.0;
-			if( sample[ 0 ] < -1.0 ) sample[ 0 ] = -1.0;
-			if( sample[ 1 ] >  1.0 ) sample[ 1 ] =  1.0;
-			if( sample[ 1 ] < -1.0 ) sample[ 1 ] = -1.0;
-
 			#ifdef WORDS_BIGENDIAN
 			int64swap( ( uint32_t * )sample );
 			int64swap( ( uint32_t * )( sample + 1 ) );
@@ -432,12 +414,6 @@ void bs2b_cross_feed_f( t_bs2bdp bs2bdp, float *sample, int n )
 
 			cross_feed_d( bs2bdp, sample_d );
 
-			/* Clipping of overloaded samples */
-			if( sample_d[ 0 ] >  1.0 ) sample_d[ 0 ] =  1.0;
-			if( sample_d[ 0 ] < -1.0 ) sample_d[ 0 ] = -1.0;
-			if( sample_d[ 1 ] >  1.0 ) sample_d[ 1 ] =  1.0;
-			if( sample_d[ 1 ] < -1.0 ) sample_d[ 1 ] = -1.0;
-
 			sample[ 0 ] = ( float )sample_d[ 0 ];
 			sample[ 1 ] = ( float )sample_d[ 1 ];
 
@@ -464,12 +440,6 @@ void bs2b_cross_feed_fbe( t_bs2bdp bs2bdp, float *sample, int n )
 
 			cross_feed_d( bs2bdp, sample_d );
 
-			/* Clipping of overloaded samples */
-			if( sample_d[ 0 ] >  1.0 ) sample_d[ 0 ] =  1.0;
-			if( sample_d[ 0 ] < -1.0 ) sample_d[ 0 ] = -1.0;
-			if( sample_d[ 1 ] >  1.0 ) sample_d[ 1 ] =  1.0;
-			if( sample_d[ 1 ] < -1.0 ) sample_d[ 1 ] = -1.0;
-
 			sample[ 0 ] = ( float )sample_d[ 0 ];
 			sample[ 1 ] = ( float )sample_d[ 1 ];
 
@@ -501,12 +471,6 @@ void bs2b_cross_feed_fle( t_bs2bdp bs2bdp, float *sample, int n )
 
 			cross_feed_d( bs2bdp, sample_d );
 
-			/* Clipping of overloaded samples */
-			if( sample_d[ 0 ] >  1.0 ) sample_d[ 0 ] =  1.0;
-			if( sample_d[ 0 ] < -1.0 ) sample_d[ 0 ] = -1.0;
-			if( sample_d[ 1 ] >  1.0 ) sample_d[ 1 ] =  1.0;
-			if( sample_d[ 1 ] < -1.0 ) sample_d[ 1 ] = -1.0;
-
 			sample[ 0 ] = ( float )sample_d[ 0 ];
 			sample[ 1 ] = ( float )sample_d[ 1 ];
 
