diff --git a/src/mkheader.c b/src/mkheader.c
index 09727d5..815dec7 100644
--- a/src/mkheader.c
+++ b/src/mkheader.c
@@ -606,7 +606,7 @@ write_special (const char *fname, int lnr, const char *tag)
     }
   else if (!strcmp (tag, "include:os-add"))
     {
-      if (!strcmp (host_os, "mingw32"))
+      if (strstr (host_os, "mingw"))
         {
           include_file (fname, lnr, "w32-add.h", write_line);
         }
