diff --git a/tools/psl.c b/tools/psl.c
index 95d8b65..85dcbe5 100644
--- a/tools/psl.c
+++ b/tools/psl.c
@@ -88,12 +88,9 @@ static void init_windows(void) {
 static const char *time2str(time_t t)
 {
 	static char buf[64];
-	struct tm tm;
+	struct tm *tp = localtime(&t);
 
-	if (localtime_r(&t, &tm) != NULL)
-		strftime(buf, sizeof(buf), "%a, %d %b %Y %H:%M:%S %Z", &tm);
-	else
-		strcpy(buf, "--notime--");
+	strftime(buf, sizeof(buf), "%a, %d %b %Y %H:%M:%S %Z", tp);
 	return buf;
 }
 
