diff -ruN libsoup-2.68.4.orig/examples/meson.build libsoup-2.68.4/examples/meson.build
--- libsoup-2.68.4.orig/examples/meson.build	2020-02-26 00:57:19.838775400 +0100
+++ libsoup-2.68.4/examples/meson.build	2020-02-29 15:13:56.081764160 +0100
@@ -9,6 +9,6 @@
   'simple-proxy'
 ]
 
-foreach example: examples
-  executable(example, example + '.c', dependencies: deps)
-endforeach
+#foreach example: examples
+  #executable(example, example + '.c', dependencies: deps)
+#endforeach
