diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6914557..926a432 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,10 +4,14 @@ project(mingw_util CXX)
 set(CMAKE_CXX_STANDARD 17)
 add_compile_options("-Wall")
 
+include_directories(${Boost_INCLUDE_DIR})
+link_directories(${Boost_LIBRARY_DIR})
+
 add_executable(peldd
   peldd.cc
   pe-parse/pe-parser-library/src/parse.cpp
   pe-parse/pe-parser-library/src/buffer.cpp
+  pe-parse/pe-parser-library/src/unicode_codecvt.cpp
   )
 set_property(TARGET peldd PROPERTY INCLUDE_DIRECTORIES
     ${CMAKE_CURRENT_SOURCE_DIR}/pe-parse/pe-parser-library/include)
diff --git a/peldd.cc b/peldd.cc
index 08ac8f2..93037f4 100644
--- a/peldd.cc
+++ b/peldd.cc
@@ -1,8 +1,8 @@
 // The code in this file is licensed under the MIT License (MIT).
 
 
-#include <parser-library/parse.h>
-#include <parser-library/nt-headers.h>
+#include <pe-parse/parse.h>
+#include <pe-parse/nt-headers.h>
 
 #include <iostream>
 #include <array>
@@ -19,6 +19,7 @@
 #include <stdexcept>
 #include <stdlib.h>
 #include <string.h>
+#include <vector>
 
 using namespace std;
 using namespace peparse;
@@ -48,11 +49,11 @@ struct section {
 };
 
 struct parsed_pe_internal {
-  list<section>   secs;
+  std::vector<section>   secs;
 };
 
 #define READ_DWORD_NULL(b, o, inst, member)                                     \
-  if (!readDword(b, o + _offset(__typeof__(inst), member), inst.member)) { \
+  if (!readDword(b, o + offsetof(__typeof__(inst), member), inst.member)) { \
     PE_ERR(PEERR_READ);                                                    \
     return nullptr;                                                          \
   }
@@ -60,11 +61,8 @@ struct parsed_pe_internal {
 
 // XXX library symbols are too generic
 extern bool getHeader(bounded_buffer *file, pe_header &p, bounded_buffer *&rem);
-extern bool getSections( bounded_buffer  *b, 
-                  bounded_buffer  *fileBegin,
-                  nt_header_32    &nthdr, 
-                  list<section>   &secs);
-extern bool getSecForVA(list<section> &secs, VA v, section &sec);
+extern bool getSections(bounded_buffer *b, bounded_buffer *fileBegin, nt_header_32 &nthdr, std::vector<section> &secs);
+extern bool getSecForVA(const std::vector<section> &secs, VA v, section &sec);
 }
 
 // most of the following function body is copied from ParsePEFromFile()
