diff --git a/src/google/protobuf/compiler/objectivec/text_format_decode_data.cc b/src/google/protobuf/compiler/objectivec/text_format_decode_data.cc
index b3083e6..a8b85f4 100644
--- a/src/google/protobuf/compiler/objectivec/text_format_decode_data.cc
+++ b/src/google/protobuf/compiler/objectivec/text_format_decode_data.cc
@@ -35,6 +35,7 @@
 #include <sstream>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #include "absl/strings/ascii.h"
 #include "absl/strings/escaping.h"
diff --git a/src/google/protobuf/compiler/objectivec/text_format_decode_data.h b/src/google/protobuf/compiler/objectivec/text_format_decode_data.h
index ab5b493..958a833 100644
--- a/src/google/protobuf/compiler/objectivec/text_format_decode_data.h
+++ b/src/google/protobuf/compiler/objectivec/text_format_decode_data.h
@@ -34,6 +34,7 @@
 #include <string>
 #include <utility>
 #include <vector>
+#include <cstdint>
 
