diff --git a/src/google/protobuf/compiler/objectivec/text_format_decode_data.cc b/src/google/protobuf/compiler/objectivec/text_format_decode_data.cc
index b3083e6..a8b85f4 100644
--- a/src/google/protobuf/compiler/objectivec/text_format_decode_data.cc
+++ b/src/google/protobuf/compiler/objectivec/text_format_decode_data.cc
@@ -35,6 +35,7 @@
 #include <sstream>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #include "absl/strings/ascii.h"
 #include "absl/strings/escaping.h"
diff --git a/src/google/protobuf/compiler/objectivec/text_format_decode_data.h b/src/google/protobuf/compiler/objectivec/text_format_decode_data.h
index ab5b493..958a833 100644
--- a/src/google/protobuf/compiler/objectivec/text_format_decode_data.h
+++ b/src/google/protobuf/compiler/objectivec/text_format_decode_data.h
@@ -34,6 +34,7 @@
 #include <string>
 #include <utility>
 #include <vector>
+#include <cstdint>
 
 // Must be included last
 #include "google/protobuf/port_def.inc"
diff --git a/src/google/protobuf/port.h b/src/google/protobuf/port.h
index 9ca45d9..6bbf1d1 100644
--- a/src/google/protobuf/port.h
+++ b/src/google/protobuf/port.h
@@ -38,6 +38,7 @@
 
 #include <cassert>
 #include <cstddef>
+#include <cstdint>
 #include <new>
 #include <string>
 #include <type_traits>
