diff --git a/src/network/socket/qhttpsocketengine.cpp b/src/network/socket/qhttpsocketengine.cpp
index 725bc213..54c67cec 100644
--- a/src/network/socket/qhttpsocketengine.cpp
+++ b/src/network/socket/qhttpsocketengine.cpp
@@ -13,6 +13,8 @@
 #if !defined(QT_NO_NETWORKPROXY)
 #include <qdebug.h>
 
+#undef option
+
 QT_BEGIN_NAMESPACE
 
 using namespace Qt::StringLiterals;
diff --git a/src/network/socket/qnativesocketengine_p.h b/src/network/socket/qnativesocketengine_p.h
index 4c185b7a..2f9ba358 100644
--- a/src/network/socket/qnativesocketengine_p.h
+++ b/src/network/socket/qnativesocketengine_p.h
@@ -37,6 +37,8 @@ QT_BEGIN_NAMESPACE
 #  define QT_SOCKOPTLEN_T int
 #endif
 
+#undef option
+
 namespace {
 namespace SetSALen {
     template <typename T> void set(T *sa, typename std::enable_if<(&T::sa_len, true), QT_SOCKLEN_T>::type len)
