diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0b783e8..caf5f66 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -83,6 +83,12 @@ target_link_libraries(qtsparkle
   ${QT_LIBRARIES}
 )
 
+if(BUILD_SHARED_LIBS)
+  target_compile_definitions(qtsparkle PRIVATE QTSPARKLE_BUILD) # dllexport
+else()
+  target_compile_definitions(qtsparkle PUBLIC QTSPARKLE_STATIC) # suppress dllimport
+endif()
+
 install(TARGETS qtsparkle
   ARCHIVE DESTINATION lib
   LIBRARY DESTINATION bin
diff --git a/src/followredirects.h b/src/followredirects.h
index aae6960..d337d91 100644
--- a/src/followredirects.h
+++ b/src/followredirects.h
@@ -38,11 +38,11 @@ public:
 
   QNetworkReply* reply() const;
 
-signals:
+Q_SIGNALS:
   void Finished();
   void RedirectLimitReached();
 
-private slots:
+private Q_SLOTS:
   void FinishedSlot();
 
 private:
diff --git a/src/uicontroller.h b/src/uicontroller.h
index 60409d9..846a1fb 100644
--- a/src/uicontroller.h
+++ b/src/uicontroller.h
@@ -44,7 +44,7 @@ public:
   void SetIcon(const QIcon& icon);
   void SetVersion(const QString& version);
 
-public slots:
+public Q_SLOTS:
   void CheckStarted();
   void UpdateAvailable(AppCastPtr appcast);
   void UpToDate();
diff --git a/src/updatechecker.h b/src/updatechecker.h
index 41b822a..508a3ec 100644
--- a/src/updatechecker.h
+++ b/src/updatechecker.h
@@ -43,13 +43,13 @@ public:
   void SetVersion(const QString& version);
   void Check(const QUrl& appcast_url, bool override_user_skip);
 
-signals:
+Q_SIGNALS:
   void CheckStarted();
   void UpdateAvailable(AppCastPtr appcast);
   void UpToDate();
   void CheckFailed(const QString& reason);
 
-private slots:
+private Q_SLOTS:
   void Finished();
   void RedirectLimitReached();
 
diff --git a/src/updatedialog.h b/src/updatedialog.h
index 88219c8..c5887a5 100644
--- a/src/updatedialog.h
+++ b/src/updatedialog.h
@@ -44,10 +44,10 @@ public:
   void SetIcon(const QIcon& icon);
   void SetVersion(const QString& version);
 
-public slots:
+public Q_SLOTS:
   void ShowUpdate(AppCastPtr appcast);
 
-private slots:
+private Q_SLOTS:
   void ReleaseNotesReady();
   void Install();
   void Skip();
diff --git a/src/updater.h b/src/updater.h
index d8d5363..bad2852 100644
--- a/src/updater.h
+++ b/src/updater.h
@@ -30,6 +30,16 @@ class QIcon;
 class QNetworkAccessManager;
 class QUrl;
 
+#ifdef QTSPARKLE_STATIC
+#  define QTSPARKLE_EXPORT
+#else
+#  ifdef QTSPARKLE_BUILD
+#    define QTSPARKLE_EXPORT Q_DECL_EXPORT
+#  else
+#    define QTSPARKLE_EXPORT Q_DECL_IMPORT
+#  endif
+#endif
+
 namespace qtsparkle {
 
 // Loads qtsparkle's translations from the .ts files compiled into the library,
@@ -49,7 +59,7 @@ void LoadTranslations(const QString& language);
 // permission, it will check for updates automatically on startup.
 // Checking for updates and displaying dialogs is done after the application
 // returns to the event loop, not in the constructor.
-class Updater : public QObject {
+class QTSPARKLE_EXPORT Updater : public QObject {
   Q_OBJECT
 
 public:
@@ -81,12 +91,12 @@ public:
   // Minimum value is one hour (3600000)
   void SetUpdateInterval(int msec);
 
-public slots:
+public Q_SLOTS:
   // Checks for updates now.  You probably want to call this from a menu item
   // in your application's main window.
   void CheckNow();
 
-private slots:
+private Q_SLOTS:
   void AutoCheck();
 
 protected:
