diff --git a/shell.c b/shell.c
index fd244bf..1a3f5b0 100644
--- a/shell.c
+++ b/shell.c
@@ -36188,7 +36188,7 @@ static int vfstraceOut(const char *z, void *pArg){
 ** UTF8, then invoke the traditional main() entry point which is
 ** renamed using a #define to utf8_main() .
 */
-#if defined(_WIN32) && !defined(main)
+#if defined(_WIN32) && !defined(__MINGW32__) && !defined(main)
 #  define main utf8_main                 /* Rename entry point to utf_main() */
 int SQLITE_CDECL utf8_main(int,char**);  /* Forward declaration */  
 int SQLITE_CDECL wmain(int argc, wchar_t **wargv){
