diff -ruN xine-lib-1.2.9.orig/configure.ac xine-lib-1.2.9/configure.ac
--- xine-lib-1.2.9.orig/configure.ac	2018-01-11 13:49:46.000000000 +0100
+++ xine-lib-1.2.9/configure.ac	2018-06-21 23:52:32.986927379 +0200
@@ -348,7 +348,7 @@
 AC_CHECK_HEADERS([alloca.h])
 AC_CHECK_HEADERS([assert.h byteswap.h dirent.h errno.h execinfo.h fcntl.h glob.h])
 AC_CHECK_HEADERS([libgen.h malloc.h netdb.h pwd.h stdbool.h ucontext.h])
-AC_CHECK_HEADERS([sys/ioctl.h sys/mixer.h sys/mman.h sys/param.h sys/socket.h sys/times.h sys/wait.h sys/sysmacros.h])
+AC_CHECK_HEADERS([sys/mixer.h sys/mman-win32.h sys/param.h sys/socket.h sys/times.h sys/wait.h sys/sysmacros.h])
 AC_CHECK_HEADERS([arpa/inet.h netinet/in.h])
 
 dnl This is duplicative due to AC_HEADER_STDC, but src/input/vcd stuff needs to
@@ -388,7 +388,7 @@
 
 AC_CACHE_CHECK([type of request parameter for ioctl()], [ac_cv_ioctl_request], [
     for ac_ioctl_request_type in "unsigned long" "int"; do
-        AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <sys/ioctl.h>
+        AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <windows.h>
                                           int ioctl(int fd, $ac_ioctl_request_type request, ...);]], [[]])],
                        [ac_cv_ioctl_request=$ac_ioctl_request_type], [])
     done
@@ -555,7 +555,7 @@
 
         if test "$WIN32_SYS" = "mingw32"; then
             WIN32_INCLUDES='-I$(top_srcdir)/win32/include'
-            LIBS="-lwinmm -lwsock32 $LIBS"
+            LIBS="-lwinmm -lwsock32 -lmman $LIBS"
             LDFLAGS="-Wl,--enable-stdcall-fixup $LDFLAGS"
             dnl iberty has been needed only in older versions
             AC_CHECK_LIB(iberty, strncomp, [GOOM_LIBS="-liberty"])
diff -ruN xine-lib-1.2.9.orig/lib/os_internal.h xine-lib-1.2.9/lib/os_internal.h
--- xine-lib-1.2.9.orig/lib/os_internal.h	2018-01-11 13:49:47.000000000 +0100
+++ xine-lib-1.2.9/lib/os_internal.h	2018-06-22 00:00:04.676476945 +0200
@@ -274,6 +274,7 @@
 #  ifdef __MINGW64__
 #    define mkdir(A, B) mkdir((A))
 #  else
+     _CRTIMP int __cdecl _mkdir(const char *_Path);
 #    define mkdir(A, B) _mkdir((A))
 #  endif
 
diff -ruN xine-lib-1.2.9.orig/src/input/input_file.c xine-lib-1.2.9/src/input/input_file.c
--- xine-lib-1.2.9.orig/src/input/input_file.c	2018-01-11 13:49:47.000000000 +0100
+++ xine-lib-1.2.9/src/input/input_file.c	2018-06-22 13:09:10.554943410 +0200
@@ -353,6 +353,10 @@
       filename = file_input_decode_uri(&(this->mrl[16]));
     else if (strncasecmp (this->mrl, "file://127.0.0.1/", 16) == 0)
       filename = file_input_decode_uri(&(this->mrl[16]));
+#ifdef __MINGW32__
+    else if (strncasecmp (this->mrl, "file:///", 8) == 0)
+      filename = file_input_decode_uri(&(this->mrl[8]));
+#endif
     else
       filename = file_input_decode_uri(&(this->mrl[5]));
   }
@@ -369,7 +373,7 @@
     } else if (errno == ENOENT) {
       _x_message(this->stream, XINE_MSG_FILE_NOT_FOUND, this->mrl, NULL);
       xine_log (this->stream->xine, XINE_LOG_MSG,
-                _("input_file: File not found: >%s<\n"), this->mrl);
+                _("input_file: File not found: >%s<\n"), filename);
     }
 
     free(filename);
